pkg/operator/app/workflowrundeps.go

```diff
@@ -56,6 +58,10 @@ var _ lifecycle.Loader = &WorkflowRunDeps{}
 var _ lifecycle.Persister = &WorkflowRunDeps{}
 
 func (wrd *WorkflowRunDeps) Load(ctx context.Context, cl client.Client) (bool, error) {
+	if _, err := wrd.Workflow.Load(ctx, cl); err != nil {
+		return false, err
+	}
+
 	return lifecycle.Loaders{
```

Comment:
You can just add this to the loaders list below:

```suggestion
	return lifecycle.Loaders{
		wrd.Workflow,
```

Also, it feels like maybe it should be wrapped in `lifecycle.RequiredLoader{}`? This won't be able to do anything if the underlying workflow object doesn't exist, right?